src/Client/ClientBase.cpp

```diff
@@ -74,18 +74,13 @@ namespace ErrorCodes
 namespace DB
 {
 
-volatile sig_atomic_t need_cancel_query = false;
-volatile sig_atomic_t query_cancelled = true;
+std::atomic<bool> catch_signal = false;
```

Comment:
Maybe I'm too paranoid (and we do not support platforms where atomics are not lock free), but I think we should ether add an assertion that `catch_signal.is_lock_free() == true` or replace it with `std::atomic_flag`.